### PR TITLE
Resolves #34: Amend last commit

### LINKED ISSUE #34: Add commenting to past commits
Currently, you are unable to see any comments on past commits within VisualGit. It would be useful to be able to view all the commit comments and then be able to add or edit new commit messages. 

These are the comments/messages entered when a commit is made.
They can be viewed in the terminal using the command, _:git log_ (shown below) 
image

The implementation of this feature is open to discussion.

### PR DESCRIPTION
**Pull request by zeshili**
This pull request is meant to implement the feature that would allow a user to amend the last commit message by replacing the old commit message with a new commit message. The "amend commit" button at the bottom left of VisualGit will only work if a user has one or more not pushed commits.

### CODE REVIEW

**Review by HDeBoever — APPROVED** [review]
The code is clean, and easy to understand.
The code could do with some more commenting, but the logic is straight forward and understandable.

After testing on my local machine, I was able to amend the most recent commit message, thereby mirroring the github functionality.  

I will approve this PR pending any potential issues that may arise in further testing.

File: app/components/footer/footer.component.html
Comment (AhmadAlsaif) [inline app/components/footer/footer.component.html]: For best practice, create an issue for the TODO then delete it from the code before merging it. to master

**Review by AhmadAlsaif — APPROVED** [review]
Does the code work?

- Yes

Did I test it locally?

- Yes tested it on my Mac

Does it perform its intended function, the logic is correct etc.

- logic is correct and works as intended 

Is the code well documented?

- yes, adequate comments were found in most implemented functions

Does it have correct linting/formatting?

- Looks good to me

Is there any redundant or duplicate code? Any hard-coded values?

- No

Can any of the code be replaced with library functions?

- Can't see any

Are good names used for methods, variables, etc.? Do they follow naming conventions?

- Yes, It makes sense 

P.S., 

1. Create an issue for the TODO and delete it 
2. Maybe delete the commented out functions from the code if it's not going to be used in the future

Other than this everything looks perfect, good work!

File: app/misc/git.ts
Comment (mdesco18) [inline app/misc/git.ts:1565]: need a `{  finally { refreshAll(repo) }` call here to update the graph

**Review by jbenning97 — APPROVED** [review]
**Does the code work? Did I test it locally? What OS am I using? Does it perform its intended function, the logic is correct etc.** - Yes, the code works for me. Testing on macOS 10.13.

**Is the code well documented? Would it be easily understood to other developers?** 

Yes

**Does it have correct linting/formatting?**

 Yes

**Is there any redundant or duplicate code? Any hard-coded values?**

 No, there were not redundant or duplicate code. 

**Can any of the code be replaced with library functions?** 

No

**Are good names used for methods, variables, etc.? Do they follow naming conventions?** 

Yes

**Were any other issues found?** 

I did not find any other issues during my testing.

**Pull request merged**
